### PR TITLE
feat: add working days component

### PR DESCRIPTION
**Pull request by Rishabh106**
working-days

- Added Working Days Component in Organization Directory

### CODE REVIEW

**Comment by punwai** [comment]
I am not a mentor but I can give you some tips that I have learnt on this PR.

**Comment by punwai** [comment]
The files you have changed are in the base directory of the project. This should not be the case. You should be adding the working days component to src/app/organization

File: working-days/working-days.component.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
+<div class="container">
```
Comment (punwai) [inline working-days/working-days.component.html]: The files for the working days component you have added are in the base directory of the project. This should not be the case. You should be adding the working days component to src/app/organization.

File: organization-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/** Angular Imports */
```
Comment (punwai) [inline organization-routing.module.ts]: You should not be adding a new organization-routing.module.ts, organization.component.html and organization.module.ts file. You should be editing the ones already created in src/app/organization.

File: working-days/working-days.component.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  <form>
```
Comment (punwai) [inline working-days/working-days.component.html]: Do not create multiple forms, create one single form which encapsulates all the fields. Also, wrap all the fields in a mat-card-content to ensure consistency in UI. See src/app/accounting-rules/create-rule/create-rule.html for reference.

File: working-days/working-days.component.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          <mat-checkbox color="primary">Sunday</mat-checkbox>
+        </tr>
+      </div>
+    </table>
+  </form>   
+</div>
+
+<div class="container" fxLayoutAlign="left">
+  <form>
+    <table style="border:0px">
```
Comment (punwai) [inline working-days/working-days.component.html]: Do not use tables for each field. Use a mat-form-field with mat-label instead to ensure consistency. Please see src/app/accounting-rules/create-rule/create-rule.html for reference.

File: working-days/working-days.component.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        <td><h4>Extend the term for Loans<br>following a Daily<br>Repayment Schedule</h4></td>
+        <td><mat-checkbox color="primary"></mat-checkbox></td><span style="padding: 390px"></span>
+      </tr>
+    </table>
+  </form>
+</div>
+
+<div class="container" fxLayoutGap="20px">
+  <button mat-raised-button color="warn" [routerLink]="['../']">Cancel</button>
+  <button mat-raised-button color="primary" [routerLink]="['../']">Submit</button>
```
Comment (punwai) [inline working-days/working-days.component.html]: The submitting function should be implemented. See src/app/accounting-rules/create-rule/create-rule.component.ts for reference.

File: working-days/working-days.component.html [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+</div>
```
Comment (punwai) [inline working-days/working-days.component.html]: Always ensure that the file ends with a new line.

File: working-days/working-days.component.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+@NgModule({
+  declarations:[
+    WorkingDaysComponent
+  ],
+imports: [
+  MatCheckboxModule,
+  MatSelectModule,
+  MatButtonModule,
+  MatFormFieldModule,
+  MatTableModule,
```
Comment (punwai) [inline working-days/working-days.component.ts]: This is a component so you should not be adding @NgModule to this. Please again see other components for reference.

File: working-days/working-days.component.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit } from '@angular/core';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatSelectModule } from '@angular/material/select';
+import { MatButtonModule } from '@angular/material/button';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatTableModule } from '@angular/material/table';
+import { Router, ActivatedRoute } from '@angular/router';
+import { NgModule } from '@angular/core';
+
+@NgModule({
```
Comment (punwai) [inline working-days/working-days.component.ts]: This is a component so you should not be adding @NgModule to this. Please again see other components for reference.

File: working-days/working-days.component.ts (hunk 7 above)
Comment (Rishabh106) [reply working-days/working-days.component.ts]: Sorry, my bad.

File: working-days/working-days.component.html (hunk 4 above)
Comment (abhaychawla) [reply working-days/working-days.component.html]: Adding on to this please don't use any inline styles as well. The `scss` file for the component is missing which should be used to implement the styles.

File: working-days/working-days.component.html [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+<div class="container" fxLayoutAlign="left">
+  <form>
+    <table style="border:0px">
+      <tr><td><h4>Payment due on<br>Non Working Days</h4></td>
+        <td><span style="padding: 37px"></span>
+      </td>
+        <td><mat-form-field>
+          <select matNativeControl required>
+            <option value="same-day">Same Day</option>
```
Comment (abhaychawla) [inline working-days/working-days.component.html]: Values should be fetched and populated by using the appropriate `API`. Don't hardcode.

File: working-days/working-days.component.html [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+</div>
+
+<div class="container" fxLayoutAlign="left">
+  <form>
+    <table style="border:0px">
+      <tr><td><h4>Payment due on<br>Non Working Days</h4></td>
+        <td><span style="padding: 37px"></span>
+      </td>
+        <td><mat-form-field>
+          <select matNativeControl required>
```
Comment (abhaychawla) [inline working-days/working-days.component.html]: Use `mat-select`.

Comment (Rishabh106) [reply working-days/working-days.component.html]: Ok, I've changed that.

File: working-days/working-days.component.html [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+      <tr>
+        <td><h4>Extend the term for Loans<br>following a Daily<br>Repayment Schedule</h4></td>
+        <td><mat-checkbox color="primary"></mat-checkbox></td><span style="padding: 390px"></span>
+      </tr>
+    </table>
+  </form>
+</div>
+
+<div class="container" fxLayoutGap="20px">
+  <button mat-raised-button color="warn" [routerLink]="['../']">Cancel</button>
```
Comment (abhaychawla) [inline working-days/working-days.component.html]: `color=warn` is not supposed to be used for Cancel button.

**Pull request closed without merging**